powerauth-webflow/src/main/java/io/getlime/security/powerauth/app/webflow/controller/HomeController.java

```diff
@@ -181,7 +185,24 @@ public String cancelAuthentication(HttpServletRequest request, HttpServletRespon
             authenticationManagementService.clearContext();
         }
 
-        return "redirect:" + redirectUri;
+        try {
```

Comment:
This is too manual - use `UriComponentsBuilder` instead.

https://docs.spring.io/autorepo/docs/spring-framework/4.3.5.RELEASE/javadoc-api/org/springframework/web/util/UriComponentsBuilder.html

You can use it like this:

```java
final String fixedCallbackUrl = UriComponentsBuilder.fromUriString(callbackUrl)
                            .queryParam("redirectId", redirectId).build()
                            .toUriString();
```